### PR TITLE
Add an option to reload all stores

### PR DESCRIPTION
**Pull request by TheFloppyBanana**
### Description of the PR
This creates a subcommand that reloads all stores.

### Changes Proposed in this Pull Request (List new items in CHANGELOG.MD)

- An `all` subcommand to reload all stores using the reload command

### Semver Classification

- [ ] This PR only includes documentation or non-code changes.
- [ ] This PR fixes a bug and does not change the (intended) framework interface.
- [X] This PR adds methods or properties to the framework interface.
- [ ] This PR removes or renames methods or properties in the framework interface.

### CODE REVIEW

File: src/commands/Admin/reload.js [hunk 1]
```diff
@@ -39,4 +40,18 @@ module.exports = class extends Command {
 		}
 	}
 
+	async all(message) {
+		const timer = new Stopwatch();
+		this.client.pieceStores.forEach(async (p) => {
```
Comment (kyranet) [inline src/commands/Admin/reload.js]:
```javascript
await this.client.pieceStores.map(store => store.loadAll().then(() => store.init()));
```

File: src/commands/Admin/reload.js [hunk 2]
```diff
@@ -39,4 +40,18 @@ module.exports = class extends Command {
...
+		await Promise.all(this.client.pieceStores.map(async (p) => {
+			await p.loadAll();
+			await p.init();
+		}));
+		if (this.client.shard) {
+			await this.client.shard.broadcastEval(`
+				if (this.shard.id !== ${this.client.shard.id}) this.client.pieceStores.map(async (p) => await p.loadAll());
+			`);
+		}
+		return message.sendMessage(`Reloaded all stores. (Took: ${timer.stop().toString()})`);
```
Comment (bdistin) [inline src/commands/Admin/reload.js]: toString() is unnecessary in a template literal, it already implicitly calls toString();

File: src/commands/Admin/reload.js [hunk 3]
```diff
@@ -39,4 +40,18 @@ module.exports = class extends Command {
...
 
+	async all(message) {
+		const timer = new Stopwatch();
+		await Promise.all(this.client.pieceStores.map(async (p) => {
+			await p.loadAll();
+			await p.init();
+		}));
+		if (this.client.shard) {
+			await this.client.shard.broadcastEval(`
+				if (this.shard.id !== ${this.client.shard.id}) this.client.pieceStores.map(async (p) => await p.loadAll());
```
Comment (bdistin) [inline src/commands/Admin/reload.js]: you also need to init all here...

File: src/commands/Admin/reload.js (hunk 2 above)
Comment (TheFloppyBanana) [reply src/commands/Admin/reload.js]: Noted, will push a change about this.

**Comment by bdistin** [comment]
also, instead of all, wouldn't it make more sense: `!reload everything`?

**Comment by Tylertron1998** [comment]
Wouldn't it make more sense just to use a default subcommand - i.e. `!reload`; because you're reloading _everything_ anyway?

File: src/commands/Admin/reload.js [hunk 4]
```diff
@@ -39,4 +40,21 @@ module.exports = class extends Command {
...
+		}));
+		if (this.client.shard) {
+			await this.client.shard.broadcastEval(`
+				if (this.shard.id !== ${this.client.shard.id}) this.client.pieceStores.map(async (store) => {
+					await store.loadAll();
+					await store.init();
+				});
+			`);
+		}
+		return message.sendMessage(`Reloaded all stores. (Took: ${timer.stop()})`);
```
[the remark this answers is not in the document]
Comment (TheFloppyBanana) [reply src/commands/Admin/reload.js]: Will do! :D

File: src/commands/Admin/reload.js [hunk 5]
```diff
@@ -22,21 +23,39 @@ module.exports = class extends Command {
...
 		try {
 			const itm = await piece.reload();
+			const timer = new Stopwatch();
 			if (this.client.shard) {
 				await this.client.shard.broadcastEval(`
 					if (this.shard.id !== ${this.client.shard.id}) this.${piece.store}.get('${piece.name}').reload();
 				`);
 			}
-			return message.sendMessage(message.language.get('COMMAND_RELOAD', itm.type, itm.name));
+			return message.sendMessage(message.language.get('COMMAND_RELOAD', itm.type, itm.name, timer.stop.toString()));
```
Comment (bdistin) [inline src/commands/Admin/reload.js]: just `timer.stop()` we don't want the string of the function source.

File: src/commands/Admin/reload.js [hunk 6]
```diff
@@ -22,21 +23,39 @@ module.exports = class extends Command {
 					if (this.shard.id !== ${this.client.shard.id}) this.${piece.name}.loadAll().then(() => this.${piece.name}.loadAll());
 				`);
 			}
-			return message.sendMessage(`${message.language.get('COMMAND_RELOAD_ALL', piece)} (Took: ${timer.stop()})`);
+			return message.sendMessage(`${message.language.get('COMMAND_RELOAD_ALL', piece, timer.stop())}`);
```
Comment (kyranet) [inline src/commands/Admin/reload.js]: Remove the template literal, it's unneded, since message.language.get returns a string.

**Comment by TheFloppyBanana** [comment]
What if the keyword should be `all` and all changes should happen in the language file, for example:
`COMMAND_RELOAD_ALL` => `COMMAND_RELOAD_STORE`,
`COMMAND_RELOAD_EVERYTHING` => `COMMAND_RELOAD_ALL`

**Comment by bdistin** [comment]
No, please do what we discussed last night. Keyword remains `everything` but is moved to the end, and is a default argument.

**Pull request merged**